tao_compiler/mlir/disc/transforms/fusion_utils_transform_based.cc

```diff
@@ -45,21 +42,49 @@ bool TransformBasedCpuFusionStrategy::isFusible(Operation* op) {
           rhsCntractingDims.size() == 1 && rhsCntractingDims[0] == 0);
 }
 
+bool TransformBasedCpuFusionStrategy::isFusible(Operation* op) {
+  return isSupportedDot(op) || isa<lmhlo::ConstantOp>(op);
+}
+
 bool TransformBasedCpuFusionStrategy::initFusionPattern(
     ShapeAnalysis& shapeAnalysis, FusionPattern& fusionPattern) {
-  Operation* inferredDominantOp = nullptr;
-  FusionType inferredFusionType = FusionType::kNone;
-
-  // TODO(wyzero): support dot fusion
-  bool allSupported =
-      llvm::all_of(fusionPattern.getOpList(),
-                   [&](Operation* op) { return this->isFusible(op); });
-  if (allSupported && fusionPattern.getOpList().size() == 1) {
-    inferredDominantOp = fusionPattern.getOpList()[0];
-    inferredFusionType = FusionType::kTransform;
+  // firstly init the fusion kind to kNone
+  fusionPattern.setDominantOp(nullptr);
+  fusionPattern.setFusionType(FusionType::kNone);
+
+  // special case for single operation.
+  if (fusionPattern.getOpList().size() == 1) {
+    Operation* op = *fusionPattern.getOpList().begin();
+    if (this->isFusible(op)) {
+      fusionPattern.setDominantOp(op);
+      fusionPattern.setFusionType(FusionType::kTransform);
+    }
+    return true;
   }
-  fusionPattern.setDominantOp(inferredDominantOp);
-  fusionPattern.setFusionType(inferredFusionType);
+
+  DenseSet<Value> dotWeights;
+  DenseSet<Operation*> supportedDotOps;
+  for (Operation* op : fusionPattern.getOpList()) {
+    // early return for the case where there are non supported ops.
+    if (!this->isFusible(op)) return true;
+    if (isSupportedDot(op)) {
+      supportedDotOps.insert(op);
+      dotWeights.insert(op->getOperand(1));
```

Comment:
nit: No need to check if it's a constant op? could operand 0 be constant (seems unusual ...)?

Reply:
the producer of `op->getOperand(1)` is usually a alloc op, not const op. The const op is also one consumer of `op->getOperand(1)`. We this at line 79.

Reply:
got it.